src/stream/src/executor/source/source_executor.rs

```diff
@@ -156,6 +156,8 @@ impl<S: StateStore> SourceExecutor<S> {
     async fn refill_row_id_column(&mut self, chunk: StreamChunk, append_only: bool) -> StreamChunk {
         let row_id_index = self.source_desc.row_id_index;
         let row_id_column_id = self.source_desc.columns[row_id_index as usize].column_id;
+        let pk_column_ids = &self.source_desc.pk_column_ids;
+        let specified_pk = pk_column_ids != &vec![0];
```

Comment:
This looks too ad-hoc. 🥵🥵🥵 What if the pk columns are really `[0]`?

Reply:
position 0 is reserved for row_id column and it is injected when registering source in meta. If user does not specify a pk, `pk_column_ids` will always be `[0]`

Reply:
Why don't we make `row_id_index` an `Option` for this?

https://github.com/singularity-data/risingwave/blob/b36f6febd0b7814a08e4afed9c0d43ee29446e41/src/source/src/manager.rs#L93-L95

Reply:
> position 0 is reserved for row_id column and it is injected when registering source in meta.

Is this documented somewhere in the code?

Reply:
> > position 0 is reserved for row_id column and it is injected when registering source in meta.
> 
> Is this documented somewhere in the code?

I think it is not documented. you can see code here https://github.com/singularity-data/risingwave/blob/2ad42f78bb17a747a03713a4bfecae94ace3df9b/src/frontend/src/handler/create_table.rs#L54

column_id always starts at 1